llvm/include/llvm/IR/ModuleSummaryIndex.h

```diff
@@ -1808,9 +1808,9 @@ class ModuleSummaryIndex {
   /// the ThinLTO backends.
   TypeIdSummary &getOrInsertTypeIdSummary(StringRef TypeId) {
     auto TidIter = TypeIdMap.equal_range(GlobalValue::getGUID(TypeId));
-    for (auto It = TidIter.first; It != TidIter.second; ++It)
-      if (It->second.first == TypeId)
-        return It->second.second;
+    for (auto &KV : make_range(TidIter))
```

Comment:
[Structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) allows you to name the pair elements directly, which I think help readability (you might know of better names than `Key`/`Value`). I'm not sure if this will work here or in the rest of this PR, but it's worth a try.

```suggestion
    for (auto &[Key, Value] : make_range(TidIter))
```

Reply:
yeah, given how the original code is already using auto anyway, I think this is a big win because it would also allow renaming each entry of the pair with a more meaningful name

Reply:
I avoided structured binding where one of the two variables is not used because I was afraid that the host compiler might issue an unused variable warning.

Now, I just tried Compiler Explorer.  gcc-7.4 issues a warning, but gcc-8.1 doesn't.  For clang, version 5 and later do not issue a warning.  (I didn't go back any further).